server/api/crud/model_monitoring/deployment.py

```diff
@@ -1271,31 +1103,23 @@ def set_credentials(
                     mlrun.common.schemas.model_monitoring.ProjectSecretKeys.STREAM_PATH
                 )
                 or mlrun.mlconf.model_endpoint_monitoring.stream_connection
-                or mlrun.mlconf.model_endpoint_monitoring.store_prefixes.stream  # TODO: Delete in 1.9.0
                 or _default_secrets_v3io
-                or old_secrets_dict.get(
-                    mlrun.common.schemas.model_monitoring.ProjectSecretKeys.STREAM_PATH
-                )  # TODO: Delete in 1.9.0
             )
-        if stream_path or stream_path == "":
+        if stream_path:
             if (
                 stream_path == mm_constants.V3IO_MODEL_MONITORING_DB
                 and mlrun.mlconf.is_ce_mode()
             ):
                 raise mlrun.errors.MLRunInvalidMMStoreTypeError(
                     "In CE mode, only kafka stream are supported for stream path"
                 )
-            elif stream_path == mm_constants.V3IO_MODEL_MONITORING_DB:
-                # TODO: Delete in 1.9.0 (for BC)
-                stream_path = ""
             elif stream_path:
                 if stream_path.startswith("kafka://") and "?topic" in stream_path:
```

Comment:
```suggestion
                elif stream_path.startswith("kafka://") and "?topic" in stream_path:
```
due to the above changes we already know that there is a stream path in this part